File: insertionSort.java
    public static void insertionSort1(int n, List<Integer> arr) {
    // Write your code here
        int last = arr.get(n - 1), current;
        
        for(int i = n - 2; i >= 0; i--) {
            current = arr.get(i);
            if(last < current) {
                arr.set(i + 1, current);
                System.out.println(arr);
            } else{
                arr.set(i + 1, last);
                System.out.println(arr);
                break;
            }
            
            if(i == 0 && current > last) {
                arr.set(i, last);
                System.out.print(arr);
            }
        }
    }

}

File: insertion_sort.py
def insertionSort1(n, arr):
    # Write your code here
    num = arr[n - 1]
    j = n - 2
    temp = arr[j]
      
    while temp >= num and j >= 0:
        arr[j + 1] = temp
        j -= 1
        temp = arr[j]
        print(*arr)
        
    arr[j + 1] = num    
    print(*arr)
